File: CalcTest.py
import unittest
from Calc import Calculator  # The class we are going to implement

class TestCalculator(unittest.TestCase):
    def test_add(self):
        calc = Calculator()
        result = calc.add(2, 3)
        self.assertEqual(result, 5)  # Expect 2 + 3 = 5

    def test_subtract(self):
        calc = Calculator()
        result = calc.subtract(5, 2)
        self.assertEqual(result, 3)  

    def test_multiply(self):
        calc = Calculator()
        result = calc.multiply(4, 3)
        self.assertEqual(result, 12)  

    def test_divide_returns_quotient(self):
        calc = Calculator()
        result = calc.divide(10, 2)
        self.assertEqual(result, 5)  

    def test_divide_handles_non_even_results(self):
        calc = Calculator()
        result = calc.divide(7, 2)
        self.assertEqual(result, 3.5)  # Expect float result for non-even division

    def test_divide_by_zero_raises_error(self):
        calc = Calculator()
        with self.assertRaisesRegex(ValueError, "Cannot divide by zero"):
            calc.divide(4, 0)  # Division by zero should not be allowed

if __name__ == "__main__":
    unittest.main()

File: calc.test.js
const test = require('node:test');
const assert = require('node:assert');

const Calc = require('./Calc');

test('Calc.add returns the sum of two integers', () => {
    assert.strictEqual(Calc.add(2, 3), 5);
});

test('Calc.subtract returns the difference of two integers', () => {
    assert.strictEqual(Calc.subtract(5, 2), 3);
});

test('Calc.multiply returns the product of two integers', () => {
    assert.strictEqual(Calc.multiply(4, 3), 12);
});

test('Calc.divide returns the quotient of two integers when divisible', () => {
    assert.strictEqual(Calc.divide(10, 2), 5);
});

test('Calc.divide returns floating point results for non-even division', () => {
    assert.strictEqual(Calc.divide(7, 2), 3.5);
});

test('Calc.divide throws when dividing by zero', () => {
    assert.throws(() => Calc.divide(4, 0), {
        name: 'Error',
        message: 'Cannot divide by zero',
    });
});
